pkg/translator/loki/logs_to_loki.go

```diff
@@ -141,6 +126,50 @@ func LogsToLokiRequests(ld plog.Logs) map[string]PushRequest {
 	return requests
 }
 
+// PushEntry is Loki log entry enriched with labels and tenant
+type PushEntry struct {
+	Entry  *push.Entry
+	Labels model.LabelSet
+	Tenant string
+}
+
+// LogToLokiEntry converts LogRecord into Loki log entry enriched with labels and tenant
+func LogToLokiEntry(lr plog.LogRecord, rl pcommon.Resource, scope pcommon.InstrumentationScope) (*PushEntry, error) {
+	// we may remove attributes, so change only our version
+	log := plog.NewLogRecord()
+	lr.CopyTo(log)
+
+	// similarly, we may remove attributes, so we make a copy and change our version
+	resource := pcommon.NewResource()
+	rl.CopyTo(resource)
+
+	// adds level attribute from log.severityNumber
+	addLogLevelAttributeAndHint(log)
+
+	format := getFormatFromFormatHint(log.Attributes(), resource.Attributes())
+	tenant := getTenantFromTenantHint(log.Attributes(), resource.Attributes())
+
+	mergedLabels := convertAttributesAndMerge(log.Attributes(), resource.Attributes())
+	// remove the attributes that were promoted to labels
+	removeAttributes(log.Attributes(), mergedLabels)
+	removeAttributes(resource.Attributes(), mergedLabels)
+
+	entry, err := convertLogToLokiEntry(log, resource, format, scope)
+	if err != nil {
+		return &PushEntry{
```

Comment:
I put an explanation above. In case of nil, tenant info will be lost. But it is needed for error report by tenant

Reply:
Is it worth using `fmt.Errorf` to include that information for you instead of providing back an incomplete PushEntry?

Reply:
I'm not sure I understand what you mean by using `fmt.Errorf`.
I can change the signature of LogToLokiEntry so it returns three values: `(string, *PushEntry, error)`. The first value is for the tenant. The `PushEntry` will not include `Tenant` field then and in case of error it will be `nil`.

Will it be better?